ui/vty-lib/src/Ariadne/UI/Vty/Widget/Wallet.hs

```diff
@@ -311,11 +313,19 @@ handleWalletWidgetEvent = \case
           walletNewAccountResultL .= NewAccountResultNone
         walletInfoL .= Just newInfo
 
-        walletNameL .= fromMaybe "" uwiLabel
+        updateEditText walletNameL walletNameEditL uwiLabel
         walletIdL .= uwiId
-        walletAccountsL .= map
-          (\(idx, UiAccountInfo{..}) -> WalletAccount idx (fromMaybe "" uaciLabel) uaciBalance False)
-          (zip [0..] uwiAccounts)
+        currentAccounts <- use walletAccountsL
+        let newAccounts = map
+              (\(idx, UiAccountInfo{..}) ->
+                WalletAccount idx (fromMaybe "" uaciLabel) uaciBalance False)
+              (zip [0..] uwiAccounts)
+            accountsEq (WalletAccount a1 a2 a3 _) (WalletAccount b1 b2 b3 _) =
+              a1 == b1 && a2 == b2 && a3 == b3
+            accountsOutdated =
+                 length currentAccounts /= length newAccounts
+              || any (not . uncurry accountsEq) (zip currentAccounts newAccounts)
```

Comment:
:cho:

Reply:
Didn't want to make `WalletAccount` an instance of `Eq` just for my use case :man_shrugging: